jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbDiagnosticsCollector.java

```diff
@@ -83,15 +86,22 @@ private void collectJsonbTransientFieldDiagnostics(ICompilationUnit unit,
         for (IType type : unit.getAllTypes()) {
             for (IField field : type.getFields()) {
                 boolean hasJsonbTransient = false, hasOtherJsonbAnnotation = false;
+                ArrayList<String> jsonbAnnotationForTheField = new ArrayList<String>();
+
                 for (IAnnotation annotation : field.getAnnotations()) {
                     String annotation_name = annotation.getElementName();
-                    if (annotation_name.equals(JsonbConstants.JSONB_TRANSIENT))
+                    
+                    if (annotation_name.equals(JsonbConstants.JSONB_TRANSIENT)) {
                         hasJsonbTransient = true;
-                    else if (annotation_name.startsWith(JsonbConstants.JSONB_PREFIX))
+                        jsonbAnnotationForTheField.add(annotation_name);
+                    } else if (annotation_name.startsWith(JsonbConstants.JSONB_PREFIX)) {
                         hasOtherJsonbAnnotation = true;
+                        jsonbAnnotationForTheField.add(annotation_name);
+                    }
                 }
                 if (hasJsonbTransient && hasOtherJsonbAnnotation) {
                     Diagnostic diagnostic = createDiagnosticBy(unit, field, JsonbConstants.JSONB_TRANSIENT);
+                    diagnostic.setData((JsonArray)(new Gson().toJsonTree(jsonbAnnotationForTheField)));
```

Comment:
I don't think we need to use Gson and JsonArray. We can just set the data to `jsonbAnnotationForTheField ` as it is for simplicity.

Reply:
The Language Server converts all `data` to a JsonArray, so this is done for it to be safe during conversion

Reply:
The Language Server converts all `data` to a JsonArray, so this is done for it to be safe during conversion